packages/ckeditor5-table/src/converters/table-heading-rows-refresh-post-fixer.js

```diff
@@ -29,14 +29,31 @@ function tableHeadingRowsRefreshPostFixer( model ) {
 	const tablesToRefresh = new Set();
 
 	for ( const change of differ.getChanges() ) {
-		if ( change.type != 'attribute' ) {
-			continue;
-		}
+		if ( change.type === 'attribute' ) {
+			const element = change.range.start.nodeAfter;
+
+			if ( element && element.is( 'element', 'table' ) && change.attributeKey === 'headingRows' ) {
+				tablesToRefresh.add( element );
+			}
+		} else {
+			/* istanbul ignore else */
```

Comment:
Alternatively instead of nested if with ignored else we can completely drop below check:

```
if ( change.type === 'insert' || change.type === 'remove' ) {
```